precovery/precovery_db.py

```diff
@@ -669,74 +472,62 @@ def precover(
         windows = self.frames.idx.window_centers(
             start_mjd, end_mjd, window_size, datasets=datasets
         )
-
         # group windows by obscodes so that many windows can be searched at once
-        matches = []
-        for obscode, obs_windows in itertools.groupby(
-            windows, key=lambda pair: pair[1]
-        ):
-            mjds = [window[0] for window in obs_windows]
-            matches_window = self._check_windows(
-                mjds,
+        candidates = []
+        frame_candidates = []
+        for obscode, mjds in windows:
+            logger.info("searching windows for obscode %s", obscode)
+            # create our Timestamp ghere
+            times = Timestamp.from_mjd(mjds, scale="utc")
+
+            candidates_obscode, frame_candidates_obscode = self._check_windows(
+                times,
                 obscode,
                 orbit,
                 tolerance,
+                propagator,
```

Comment:
```suggestion
                prop,
```

Here we send in the instance.